scaaml/io/dataset.py

```diff
@@ -162,7 +162,8 @@ def __init__(
 
         # current shard tracking
         self.shard_key: Optional[str] = None
-        self.prev_shard_key = None  # track key change for counting
+        self.prev_shard_key: Optional[
+            str] = None  # track key change for counting
```

Comment:
For readability: move the comment above the line so that the code isn't split

Reply:
Done